packages/ui/client/composables/explorer/tree.ts

```diff
@@ -0,0 +1,594 @@
+      timeString: '',
+      filesFailed: 0,
+      filesSuccess: 0,
+      filesIgnore: 0,
+      filesRunning: 0,
+      filesSkipped: 0,
+      filesTodo: 0,
+      filesSnapshotFailed: 0,
+      testsFailed: 0,
+      testsSuccess: 0,
+      testsIgnore: 0,
+      testsSkipped: 0,
+      testsTodo: 0,
+      totalTests: 0,
+    }
+    let time = 0
+    for (const [id, f] of useFiles) {
+      if (!f)
+        continue
+      const file = filesMap.get(id)
+      if (file) {
+        file.mode = f.mode
+        file.setupDuration = f.setupDuration
+        file.prepareDuration = f.prepareDuration
+        file.environmentLoad = f.environmentLoad
+        file.collectDuration = f.collectDuration
+        file.duration = f.result?.duration
+        file.state = f.result?.state
+      }
+      time += Math.max(0, f.collectDuration || 0)
+      time += Math.max(0, f.setupDuration || 0)
+      time += Math.max(0, f.result?.duration || 0)
+      time += Math.max(0, f.environmentLoad || 0)
+      time += Math.max(0, f.prepareDuration || 0)
+      data.timeString = time > 1000 ? `${(time / 1000).toFixed(2)}s` : `${Math.round(time)}ms`
+      if (f.result?.state === 'fail') {
+        data.filesFailed++
+      }
+      else if (f.result?.state === 'pass') {
+        data.filesSuccess++
+      }
+      else if (f.mode === 'skip') {
+        data.filesIgnore++
+        data.filesSkipped++
+      }
+      else if (f.mode === 'todo') {
+        data.filesIgnore++
+        data.filesTodo++
+      }
+      else {
+        data.filesRunning++
+      }
+
+      const tests = getTests(f)
+
+      data.totalTests += tests.length
+
+      for (const t of tests) {
+        if (t.result?.state === 'fail') {
+          data.testsFailed++
+        }
+        else if (t.result?.state === 'pass') {
+          data.testsSuccess++
+        }
+        else if (t.mode === 'skip') {
+          data.testsIgnore++
+          data.testsSkipped++
+        }
+        else if (t.mode === 'todo') {
+          data.testsIgnore++
+          data.testsTodo++
+        }
+      }
+    }
+    // eslint-disable-next-line no-console
+    console.log(`collect took: ${performance.now() - now}ms`)
+    this.summary.files = data.files
+    this.summary.time = data.timeString
+    this.summary.filesFailed = data.filesFailed
+    this.summary.filesSuccess = data.filesSuccess
+    this.summary.filesIgnore = data.filesIgnore
+    this.summary.filesRunning = data.filesRunning
+    this.summary.filesSkipped = data.filesSkipped
+    this.summary.filesTodo = data.filesTodo
+    this.summary.testsFailed = data.testsFailed
+    this.summary.testsSuccess = data.testsSuccess
+    this.summary.testsFailed = data.testsFailed
+    this.summary.testsTodo = data.testsTodo
+    this.summary.testsIgnore = data.testsIgnore
+    this.summary.testsSkipped = data.testsSkipped
+    this.summary.totalTests = data.totalTests
+    // update filtered files
+    filteredFiles.value = this.filteredFiles()
```

Comment:
try to add new tasks here to the tree: if we have a filter applied while running, all missing subtasks will be loaded once `onFinished` called